src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -0,0 +1,304 @@
+import { InputOperation } from "./type/InputOperation.js";
+import { parseHttpRequestMethod } from "./type/RequestMethod.js";
+import { BodyMediaType } from "./type/BodyMediaType.js";
+import { InputParameter } from "./type/InputParameter.js";
+import { InputType } from "./type/InputType.js";
+import { InputTypeKind } from "./type/InputTypeKind.js";
+import { RequestLocation, requestLocationMap } from "./type/RequestLocation.js";
+import { OperationResponse } from "./type/OperationResponse.js";
+import { getInputType } from "./lib/model.js";
+import { InputOperationParameterKind } from "./type/InputOperationParameterKind.js";
+import { resolveServers } from "./lib/cadlServer.js";
+import { getExternalDocs } from "./lib/decorators.js";
+
+export interface NetEmitterOptions {
+    outputFile: string;
+    logFile: string;
+}
+
+const defaultOptions = {
+    outputFile: "cadl.json",
+    logFile: "log.json"
+};
+
+export async function $onEmit(program: Program, emitterOptions: NetEmitterOptions) {
+    const resolvedOptions = {...defaultOptions, ...emitterOptions };
+    const options: NetEmitterOptions = {
+        outputFile: resolvePath(
+            program.compilerOptions.outputPath ?? "./cadl-output",
+            resolvedOptions.outputFile
+        ),
+        logFile: resolvePath(
+            program.compilerOptions.outputPath ?? "./cadl-output",
+            resolvedOptions.logFile
+        ),
+    };
+    const version: string = "";
+    if (!program.compilerOptions.noEmit && !program.hasError()) {
+        // Write out the dotnet model to the output path
+        const namespace = getServiceNamespaceString(program)?.toLowerCase() || "";
+        const outPath = version.trim().length > 0 
+            ? resolvePath(options.outputFile?.replace(".json", `.${version}.json`))
+            : resolvePath(options.outputFile);
+        
+        const root = createModel(program);
+        // await program.host.writeFile(outPath, prettierOutput(JSON.stringify(root, null, 2)));
+        await program.host.writeFile(outPath, prettierOutput(stringifyRefs(root, null, 1, PreserveType.Objects)));
+        const yamlOutPath = resolvePath(options.outputFile?.replace(".json", `.yaml`));
+        await program.host.writeFile(yamlOutPath, dump(root));
+      }
+}
+
+function prettierOutput(output: string) {
+    return output + "\n";
+}
+function getClient(clients: InputClient[], clientName:string): InputClient|undefined {
+    for (const client of clients) {
+        if (client.Name == clientName) return client;
+    }
+
+    return undefined;
+}
+
+function createModel(program: Program): any {
+    const serviceNamespaceType = getServiceNamespace(program);
+    if (!serviceNamespaceType) {
+      return;
+    }
+    const title = getServiceTitle(program);
+    const version = getServiceVersion(program);
+    if (version === "0000-00-00") {
+        console.error("No API-Version provided.")
```

Comment:
Is it better to continue processing CADL and let codegen to report error?

Reply:
How could we gaurantee it must be returning "0000-00-00" in all the locales.

Reply:
getServiceVersion api will return "0000-00-00" when no service information is provided.